vllm/model_executor/layers/fused_moe/layer.py

```diff
@@ -1251,15 +1260,37 @@ def select_experts(
         if use_grouped_topk:
             assert topk_group is not None
             assert num_expert_group is not None
-            topk_weights, topk_ids = grouped_topk(
-                hidden_states=hidden_states,
-                gating_output=router_logits,
-                topk=top_k,
-                renormalize=renormalize,
-                num_expert_group=num_expert_group,
-                topk_group=topk_group,
-                scoring_func=scoring_func,
-                e_score_correction_bias=e_score_correction_bias)
+            if hidden_states.shape[0] == 0:
+                topk_ids = torch.full((0, top_k),
+                                      -1,
+                                      dtype=torch.int,
+                                      device=hidden_states.device)
+                topk_weights = torch.empty((0, top_k),
+                                           dtype=torch.float32,
+                                           device=hidden_states.device)
```

Comment:
Please include a comment for this new case, I assume this can happen during DP/EP? cc @varun-sundar-rabindranath @tlrmchlsmth